src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -2630,16 +2656,19 @@ class Simplex_tree {
   /** \brief Total number of simplices in the complex, without the empty simplex.*/
   /** \brief Set of simplex tree Nodes representing the vertices.*/
   Siblings root_;
+
+  // all mutable as their content has no impact on the content of the simplex tree itself
+  // they correspond to some kind of cache or helper attributes.
```

Comment:
At some point we should document what functions are safe to use in parallel on a Simplex_tree and what functions are not. Some people have the expectation that `const` means thread-safe, which isn't the case in the presence of `mutable`.

Reply:
I marked `dimension()`, `upper_bound_dimension()`, `initialize_filtration()`, `maybe_initialize_filtration ()`, `clear_filtration()`, `filtration_simplex_range()`, `simplex(Simplex_key idx)`, `print_hasse()` as not thread-safe in the doc on 3d21b0cb2
Remark, that `print_hasse()` should not and use `complex_simplex_range()` instead of `filtration_simplex_range()`.